test_util/test_upgrade_vpc.py

```diff
@@ -38,30 +38,68 @@
     This value will be used when upgrading the cluster.
 
 """
-import collections
-import datetime
 import logging
 import os
+import pprint
 import random
 import string
 import sys
 import uuid
+from contextlib import contextmanager
+
+import retrying
 
 import test_util.aws
 import test_util.cluster
 from pkgpanda.util import load_string
 from test_util.cluster_api import ClusterApi
-from test_util.helpers import CI_AUTH_JSON, DcosUser
-from test_util.marathon import TEST_APP_NAME_FMT
+from test_util.helpers import CI_AUTH_JSON, DcosUser, marathon_app_id_to_mesos_dns_subdomain
 
 
 logging.basicConfig(format='[%(asctime)s|%(name)s|%(levelname)s]: %(message)s', level=logging.DEBUG)
 log = logging.getLogger(__name__)
 
 
-def get_test_app():
-    app = {
-        "id": TEST_APP_NAME_FMT.format(uuid.uuid4().hex),
+TEST_APP_NAME_FMT = 'upgrade-{}'
+
+
+@contextmanager
+def cluster_workload(cluster_api):
+    """Start a cluster workload on entry and verify its health on exit."""
+
+    def app_task_ids(app_id):
+        """Return a list of Mesos task IDs for app_id's running tasks."""
+        assert app_id.startswith('/')
+        response = cluster_api.marathon.get('/v2/apps' + app_id + '/tasks')
+        response.raise_for_status()
+        tasks = response.json()['tasks']
+        return [task['id'] for task in tasks]
+
+    def parse_dns_log(dns_log_content):
```

Comment:
Yeah, I find that code separation has benefits aside from reusability, namely isolation. It's nice to be able to see how DNS availability is verified without getting into the details of how the DNS resolution log is structured and parsed. And if I do want to know about that, it's helpful to be able to reason about the log file's format without getting into how it's used.